rest_framework_jwt/serializers.py

```diff
@@ -55,12 +55,29 @@ def validate(self, attrs):
                     msg = _('User account is disabled.')
                     raise serializers.ValidationError(msg)
 
-                payload = jwt_payload_handler(user)
-
-                return {
-                    'token': jwt_encode_handler(payload),
-                    'user': user
-                }
+                data = {}
+                device = None
+                if api_settings.JWT_PERMANENT_TOKEN_AUTH:
+                    headers = self.context['request'].META
+                    device_name = headers.get('HTTP_X_DEVICE_MODEL')
+                    user_agent = headers.get('HTTP_USER_AGENT')
+                    if not device_name:
+                        device_name = user_agent
+                        device_details = ''
+                    else:
+                        device_details = user_agent
+
+                    device = Device.objects.create(
+                        user=user, last_request_datetime=datetime.now(),
+                        name=device_name, details=device_details)
+                    data['device'] = device
+
+                payload = jwt_payload_handler(user, device=device)
+                data.update({
```

Comment:
Nope, the url is correct, I meant to show the old "return", but I hope you see my point.